File: backend/Models/Message.cs
using System;

namespace RentAPlace.Models
{
    public class Message
    {
        public int Id { get; set; }

        public int SenderId { get; set; }

        public int ReceiverId { get; set; }

        public int PropertyId { get; set; }

        public string MessageText { get; set; } = "";

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public User? Sender { get; set; }

        public User? Receiver { get; set; }

        public Property? Property { get; set; }
    }
}

File: frontend/src/app/services/message.ts
import { Injectable } from '@angular/core';
import { timeout } from 'rxjs';

import { ApiService } from './api.service';

export interface AppUserLite {
  id: number;
  name?: string;
  email?: string;
  role?: string;
}

export interface AppPropertyLite {
  id: number;
  title?: string;
  location?: string;
  propertyType?: string;
  pricePerNight?: number;
  ownerId?: number;
}

export interface AppMessage {
  id: number;
  senderId: number;
  receiverId: number;
  propertyId: number;
  messageText: string;
  createdAt: string;
  sender?: AppUserLite;
  receiver?: AppUserLite;
  property?: AppPropertyLite;
}

export interface SendMessageRequest {
  receiverId: number;
  propertyId: number;
  messageText: string;
}

const MESSAGES_TIMEOUT_MS = 12000;

@Injectable({
  providedIn: 'root'
})
export class MessageService {
  private baseUrl = '/api/Message';

  constructor(private api: ApiService) {}

  getMessages() {
    return this.api.get<AppMessage[]>(this.baseUrl).pipe(timeout(MESSAGES_TIMEOUT_MS));
  }

  sendMessage(payload: SendMessageRequest) {
    return this.api.post<AppMessage>(this.baseUrl, payload).pipe(timeout(MESSAGES_TIMEOUT_MS));
  }
}

